imjoy/engine.py

```diff
@@ -50,8 +51,21 @@ def setup(self):
         setup_services(self)
         setup_runners(self)
 
+    def start_proxy(self):
+        """Start the reverse proxy for public access."""
+        try:
+            from pyngrok import ngrok
+
+            public_url = ngrok.connect(port=self.opt.port)
+            public_url = public_url.replace("http", "https")
+            self.public_url = public_url
+        except:
```

Comment:
We should except the pyngrok error here:
https://pyngrok.readthedocs.io/en/latest/api.html#pyngrok.exception.PyngrokError

Reply:
Yes, I can add that one. However, I am not confident enough to make sure that's the only error type can happen. We cannot afford that crashing the engine because the proxy failed to start. For example `import pyngrok` may also fail.

Reply:
Then make two except cases.

Reply:
Would you do something like this?
```python
        try:
            from pyngrok import ngrok
            from pyngrok.exception import PyngrokError
            try:
                public_url = ngrok.connect(port=self.opt.port)
                public_url = public_url.replace("http", "https")
                self.public_url = public_url
            except PyngrokError:
                self.logger.error("ngrok failed to start.")
        except ImportError:
            self.logger.error("Failed to import pyngrok")
```

Reply:
Well, perhaps in this case,  I can just remove the try block to let it crash, the user can turn of the feature with the `--public` anyway.

Reply:
When would there be an import error?

I would return early if we can't import pyngrok.

But it would be better if we knew earlier that pyngrok is installed.